### PR TITLE
[Perf] Ming-TTS Stage-0 flow-head speedup: RoPE/QKV fusion, opt-in torch.compile, PIECEWISE cudagraph

### PR DESCRIPTION
**Pull request by ZhengWG**
PLEASE FILL IN THE PR DESCRIPTION HERE.

## Purpose

Perf for ming-tts model.

## Test Plan

**vLLM Version:** v0.24.0

**vLLM-Omni Commit:** 3e578178bd989e15e886027dc43173b8dd1359c0

## Test Result

serve script:
```bash
vllm serve "$MODEL" \
    --deploy-config "$DEPLOY_CONFIG" \
    --host "$HOST" \
    --port "$PORT" \
    --trust-remote-code \
    --omni
```

bench script:
```bash
vllm bench serve --omni \
    --host "$HOST" --port "$PORT" \
    --model "$MODEL" \
    --trust-remote-code \
    --backend openai-audio-speech \
    --endpoint /v1/audio/speech \
    --dataset-name seed-tts \
    --dataset-path BytedanceSpeech/seed-tts-eval \
    --seed-tts-locale en \
    --num-prompts 20 \
    --num-warmups 2 \
    --hf-output-len 200 \
    --max-concurrency 1 \
    --request-rate inf \
    --percentile-metrics ttft,e2el,audio_rtf,audio_ttfp,audio_duration,audio_underrun \
    --save-result \
    --result-dir "$RESULT_DIR"
```

Test on `seed-tts-eval` dataset: 

```bash
vllm bench serve --omni \
    --host "$HOST" --port "$PORT" \
    --model "$MODEL" \
    --trust-remote-code \
    --backend openai-audio-speech \
    --endpoint /v1/audio/speech \
    --dataset-name seed-tts \
    --dataset-path "$DATASET_PATH" \
    --seed-tts-locale en \
    --num-prompts 1100 \
    --num-warmups 2 \
    --max-concurrency 16 \
    --request-rate inf \
    --percentile-metrics ttft,e2el,audio_rtf,audio_ttfp,audio_duration,audio_underrun \
    --save-result \
    --result-dir "$RESULT_DIR"
```

bench result: 

| Feature              | E2EL mean | Step Δ | Cumulative Δ | TTFP mean | RTF  | Bench time |
|----------------------|-----------|--------|--------------|-----------|------|------------|
| baseline             | 1428.0 ms | —      | —            | 1424.0 ms | 0.19 | 28.57 s    |
| + RoPE cache         | 1422.5 ms | −0.4%  | −0.4%        | 1418.4 ms | 0.19 | 28.46 s    |
| + Fused-QKV          | 1387.9 ms | −2.4%  | −2.8%        | 1382.8 ms | 0.18 | 27.77 s    |
| + torch.compile      | 1300.3 ms | −6.3%  | −8.9%        | 1296.2 ms | 0.17 | 26.01 s    |
| + PIECEWISE (final)  | 1007.2 ms | −22.5% | −29.5%       | 1003.4 ms | 0.14 | 20.15 s    |

Seed-tts result on baseline:

```bash
============ Serving Benchmark Result ============
Successful requests:                     1100
Failed requests:                         0
Maximum request concurrency:             16
Benchmark duration (s):                  781.15
Request throughput (req/s):              1.41
Peak concurrent requests:                18.00
----------------End-to-end Latency----------------
Mean E2EL (ms):                          11283.05
Median E2EL (ms):                        11219.47
P99 E2EL (ms):                           13075.90
================== Text Result ===================
Total input tokens:                      147573
Total generated tokens:                  0
Output token throughput (tok/s):         0.00
Peak output token throughput (tok/s):    16.00
Peak concurrent requests:                18.00
Total Token throughput (tok/s):          188.92
================== Audio Result ==================
Total audio duration generated(s):       9113.41
Total audio frames generated:            218721888
Audio throughput(audio duration/s):      11.67
Streaming continuity OK rate:            100.00%
-----------------Real Time Factor-----------------
Mean AUDIO_RTF:                          1.47
Median AUDIO_RTF:                        1.37
P99 AUDIO_RTF:                           3.11
---------------Time to First Packet---------------
Mean AUDIO_TTFP (ms):                    11280.24
Median AUDIO_TTFP (ms):                  11216.33
P99 AUDIO_TTFP (ms):                     13074.34
------------------Audio Duration------------------
Mean AUDIO_DURATION (s):                 8.28
Median AUDIO_DURATION (s):               8.23
P99 AUDIO_DURATION (s):                  14.11
-------------Streaming Audio Underrun-------------
Mean AUDIO_UNDERRUN (s):                 0.00
Median AUDIO_UNDERRUN (s):               0.00
P99 AUDIO_UNDERRUN (s):                  0.00
==================================================
```

Seed-tts result on the current PR:

```bash
============ Serving Benchmark Result ============
Successful requests:                     1100
Failed requests:                         0
Maximum request concurrency:             16
Benchmark duration (s):                  606.98
Request throughput (req/s):              1.81
Peak concurrent requests:                25.00
----------------End-to-end Latency----------------
Mean E2EL (ms):                          8785.30
Median E2EL (ms):                        8767.27
P99 E2EL (ms):                           11279.72
================== Text Result ===================
Total input tokens:                      147573
Total generated tokens:                  0
Output token throughput (tok/s):         0.00
Peak output token throughput (tok/s):    16.00
Peak concurrent requests:                25.00
Total Token throughput (tok/s):          243.13
================== Audio Result ==================
Total audio duration generated(s):       9160.45
Total audio frames generated:            219850848
Audio throughput(audio duration/s):      15.09
Streaming continuity OK rate:            100.00%
-----------------Real Time Factor-----------------
Mean AUDIO_RTF:                          1.15
Median AUDIO_RTF:                        1.06
P99 AUDIO_RTF:                           2.44
---------------Time to First Packet---------------
Mean AUDIO_TTFP (ms):                    8779.47
Median AUDIO_TTFP (ms):                  8764.00
P99 AUDIO_TTFP (ms):                     11271.83
------------------Audio Duration------------------
Mean AUDIO_DURATION (s):                 8.33
Median AUDIO_DURATION (s):               8.23
P99 AUDIO_DURATION (s):                  14.11
-------------Streaming Audio Underrun-------------
Mean AUDIO_UNDERRUN (s):                 0.00
Median AUDIO_UNDERRUN (s):               0.00
P99 AUDIO_UNDERRUN (s):                  0.00
==================================================
```

**BEFORE SUBMITTING:** read [CONTRIBUTING.md](https://github.com/vllm-project/vllm-omni/blob/main/CONTRIBUTING.md) and run the [precheck-pr skill](https://github.com/vllm-project/vllm-omni/blob/main/.claude/skills/precheck-pr/SKILL.md) with the code agent for a self-check against project conventions.
(anything written below this line will be removed by GitHub Actions)

### CODE REVIEW

**Comment by ZhengWG** [comment]
@yuanheng-zhao PTLA~

File: vllm_omni/model_executor/models/ming_tts/ming_tts_llm.py [hunk 1]
```diff
@@ -326,6 +326,20 @@ def _maybe_build_cfm_graph(self, z_diff_cond: torch.Tensor):
...
+            # TODO(perf):
+            #   (1) compile per-block (dit_model.blocks) instead of the
+            #   whole forward for tighter fusion / fewer graph breaks;
+            #   (2) move the compile + a synthetic warmup forward to a post-weight-load hook
+            dit_model = self.flowloss.cfm.model
+            if not getattr(dit_model, "_ming_compiled", False):
+                try:
+                    torch._inductor.config.triton.cudagraphs = False
+                    dit_model.forward = torch.compile(dit_model.forward, fullgraph=False, dynamic=False)
+                    dit_model._ming_compiled = True
```
Comment (yuanheng-zhao) [inline vllm_omni/model_executor/models/ming_tts/ming_tts_llm.py]: Shall we better apply scoped triton.cudagraphs options like that in `_voxcpm2_compile_without_inductor_cudagraphs `?

https://github.com/vllm-project/vllm-omni/blob/ca0ae7269ca3e9487645cf66088fdfc338951da9/vllm_omni/model_executor/models/voxcpm2/voxcpm2_talker.py#L1129-L1138

File: vllm_omni/model_executor/models/common/ming/dit.py [hunk 2]
```diff
@@ -66,12 +79,32 @@ def __init__(
...
+        self._qkv_w: torch.Tensor | None = None
+        self._qkv_b: torch.Tensor | None = None
+
+    def _qkv(self, x):
+        if self._qkv_w is None:
+            self._qkv_w = torch.cat([self.to_q.weight, self.to_k.weight, self.to_v.weight], dim=0)
+            if self.to_q.bias is not None:
+                self._qkv_b = torch.cat([self.to_q.bias, self.to_k.bias, self.to_v.bias], dim=0)
+        qkv = F.linear(x, self._qkv_w, self._qkv_b)
+        return qkv.split(self.inner_dim, dim=-1)
```
Comment (yuanheng-zhao) [inline vllm_omni/model_executor/models/common/ming/dit.py:95]: Might leave a todo that we might want to materialize the fused weight eagerly outside any compiled scope. For now the lazy build might be confused with its states between any compile/re-compiles.

**Review by yuanheng-zhao — COMMENTED** [review]
Generally LGTM, left some comments as well.

Shall we also paste the comparison of WER (with `--seed-tts-wer-eval`) before and with this PR?

And a tiny note this PR is for Ming-omni-tts rather than Ming-flash-omni-2.0 (We'll update the title/scope/tasks of #4129)

**Comment by ZhengWG** [comment]
> Generally LGTM, left some comments as well.
> 
> Shall we also paste the comparison of WER (with `--seed-tts-wer-eval`) before and with this PR?
> 
> And a tiny note this PR is for Ming-omni-tts rather than Ming-flash-omni-2.0 (We'll update the title/scope/tasks of #4129)


Thank you for the valuable feedback! All comments have been addressed.

Regarding the WER benchmark, I've added the comparison results (via --seed-tts-wer-eval), which demonstrate a ~30% improvement in E2E inference time with a similar output length.

**Review by yuanheng-zhao — APPROVED** [review]
LGTM

**Comment by ZhengWG** [comment]
@yuanheng-zhao Hello. Do we need to handle the NPU/AMD CI errors? They seem unrelated to this PR.

**Comment by linyueqian** [comment]
thanks I have merged the PR

**Comment by yuanheng-zhao** [comment]
NPU/AMD CI errors look unrelated with changes in this PR
